MetaMask/metamask-extension
# Replacing setInterval with chrome.alarm in Detect-Token controller

#15509 
As a part of MV3, we need to replace the `setTimeout` and `setInterval` functions with chrome Alarms throughout the code base. While creating chrome alarms, we can define `delayInMinutes` to show how many minutes we should wait before the first time the alarm is triggered; we can also define `periodInMinutes ` which is the number of minutes between alarms. Unlike `setIntervals` the time interval has to be mentioned in Minutes, not in milliseconds. Inside the `onAlarm` event listener, we can define the task to be executed during the alarm defined.


Manual Testing:

1. Log in to the MetaMask application; the detectNewtoken() function should be triggered once every 3 mins using breakpoints (For testing purposes you can locally update the `DEFAULT_INTERVAL_MV3` value to a lower value).
2.  It should clear the existing alarm and create a new alarm every time the user switch address, switch the network, and toggle `Enhanced token detection`.
